code/clement/django/lab03/django_pokedex/pokedex_app/views.py

```diff
@@ -0,0 +1,34 @@
+from django.shortcuts import render
+from .models import PokemonType, Pokemon
+from django.db.models import Q
+
+
+# Create your views here.
+def index(request):
+    pokemons = Pokemon.objects.all()
+
+    # Applying various filters base on my form data
+    # And pull values out from the form data
+
+    pokemon_search = request.POST.get('pokemon-search') or ''
+
+    if pokemon_search:
+        # Using Q object to search within the name & types fields for pokemon_search
+        pokemons = Pokemon.objects.filter(
+            Q(name__icontains=pokemon_search) | Q(
+                types__name__icontains=pokemon_search)
+        )
+
+    form_data = {
+        'pokemon_search': pokemon_search,
+
+    }
+
+    context = {
+        'pokemon_search': pokemon_search,
+        'pokemons': pokemons,
+        'form_data': form_data,
+
+    }
```

Comment:
It is only necessary to forward the `form_data` through the `context` if it is needed in the template to populate the form values. The form fields aren't currently being populated when the filter form is rendered again, so we can either
1. Remove all references to `form_data`
2. Populate the input search box with the value

Personally, I would probably choose option 1 in this case since there aren't any additional form fields that need populating.